src/mlio/text_line_reader.cxx

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You
+ * may not use this file except in compliance with the License. A copy of
+ * the License is located at
+ *
+ *      http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is
+ * distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+ * ANY KIND, either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+
+#include "mlio/text_line_reader.h"
+
+#include <cstddef>
+#include <stdexcept>
+#include <vector>
+
+#include "mlio/cpu_array.h"
+#include "mlio/data_type.h"
+#include "mlio/instance.h"
+#include "mlio/instance_batch.h"
+#include "mlio/intrusive_ptr.h"
+#include "mlio/parallel_data_reader.h"
+#include "mlio/record_readers/text_line_record_reader.h"
+#include "mlio/schema.h"
+#include "mlio/streams/utf8_input_stream.h"
+#include "mlio/text_encoding.h"
+
+namespace mlio {
+inline namespace v1 {
+
+text_line_reader::text_line_reader(data_reader_params rdr_prm)
+    : parallel_data_reader{std::move(rdr_prm)}
+{}
+
+text_line_reader::~text_line_reader()
+{
+    stop();
+}
+
+intrusive_ptr<record_reader>
+text_line_reader::make_record_reader(data_store const &ds)
+{
+    auto strm = make_utf8_stream(ds.open_read(), text_encoding::utf8);
+    auto rdr =
+        make_intrusive<detail::text_line_record_reader>(std::move(strm), true);
```

Comment:
Not sure if we should hard-code skipping blank lines. Ideally this should be configurable, but for this first release, I would say setting this to `false` is a safer approach.